### ISSUE TITLE
Implement "goto type definition"

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#2859]
### Discussed in https://github.com/microsoft/pylance-release/discussions/2618

<div type='discussions-op-text'>

<sup>Originally posted by **tekumara** November 15, 2020</sup>
Is it possible to go to an inferred type definition?

eg: 
In this example I'm hovering over the `encryption` variable and can see the type is `GetBucketEncryptionOutputTypeDef`.
I'd like to go to the definition of the type, so I can see it in all its glory.

<img width="946" alt="Screen Shot 2020-11-16 at 10 50 17 am" src="https://user-images.githubusercontent.com/125105/99200490-bca1ea80-27f9-11eb-939e-7e3145aadb38.png">
 </div>

**Comment by erictraut** [comment]
This is already implemented, and it's working fine for me. Is it not working for you?

**Comment by debonte** [comment]
@erictraut, yes this has been implemented for a while. However, I noticed that we had an [active discussion](https://github.com/microsoft/pylance-release/discussions/2618) about it. We recently started using discussions to collect votes on enhancements and this was the first time we've had one that needed to be closed. So @judej and I were discussing how to handle that scenario. Basically, do we use GitHub's "Create issue from discussion" feature, or move the conversation back to the [original issue](https://github.com/microsoft/pylance-release/issues/611), or ...?

This was an experiment to see what "Create issue from discussion" does. There are a couple downsides:
1. The issue is shown as created by me rather than the user who originally reported the issue.
2. Now we have another issue about the same thing.
3. Apparently, based on your reply, it has the potential to confuse people. :) But I think in this case, it's because the work to implement this wasn't done recently.

On the other hand, for enhancement discussions that are the result of multiple issues, creating a single, new issue to track the work that was decided upon would make sense. And it's simplest to have the same process for all discussions regardless of how many source issues were involved.

Anyhow, the next step with this issue is to close it and let the user know what build it was implemented in.
